src/ray/raylet/main.cc

```diff
@@ -202,10 +118,110 @@ int main(int argc, char *argv[]) {
   }
   RAY_CHECK_OK(gcs_client->Connect(main_service));
 
-  std::unique_ptr<ray::raylet::Raylet> server(new ray::raylet::Raylet(
-      main_service, raylet_socket_name, node_ip_address, redis_address, redis_port,
-      redis_password, node_manager_config, object_manager_config, gcs_client));
-  server->Start();
+  // Keep server reference outside of the callback
+  std::unique_ptr<ray::raylet::Raylet> server(nullptr);
+
+  RAY_CHECK_OK(gcs_client->Nodes().AsyncGetInternalConfig([&](const std::unordered_map<
+                                                              std::string, std::string>
+                                                                  stored_raylet_config) {
+    // Parse the configuration list.
+    std::istringstream config_string(config_list);
+    std::string config_name;
+    std::string config_value;
+
+    while (std::getline(config_string, config_name, ',')) {
```

Comment:
We should only be doing this in the case of the head node now, right? I think this would be clearer if above, before the callback, we parse and set this in the GCS if it's the head node. Then when initializing the raylet, we just always pull the config from the GCS. Then you could also remove the "set_internal_config" flag to the raylet constructor.

Reply:
Okay--I'll store something if the `config_string` passed in is greater than zero.

Reply:
SGTM please ping me when you've updated